plugins/device/deviceutils/store-utils.go

```diff
@@ -51,14 +52,52 @@ func GetDevicesFromPeer(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
 	return deviceInfo, nil
 }
 
+// SetDeviceState sets device state and updates device state in etcd
+func SetDeviceState(peerID string, deviceName string, deviceState string) error {
+
+	devices, err := GetDevices(peerID)
+	if err != nil {
+		return err
+	}
+
+	index := DeviceInList(deviceName, devices)
+	if index != -1 {
```

Comment:
suggestion:
always do error check in the beginning 
```
index := DeviceInList(deviceName, devices)
if index <0{
      return errors.New("device doesnot exists in the given peer")
}else{
	devices[index].State = deviceState
	err := updateDevices(peerID, devices)
	if err != nil {
		return err
	}
} 
```

Reply:
else can be removed.

    index := DeviceInList(deviceName, devices)
    if index <0{
          return errors.New("device doesnot exists in the given peer")
    }
    devices[index].State = deviceState
    err := updateDevices(peerID, devices)
    if err != nil {
        return err
    }

Reply:
we can directly return err instead of checking it.